omod/src/main/java/org/openmrs/web/controller/PortletController.java

```diff
@@ -190,7 +193,10 @@ public ModelAndView handleRequest(HttpServletRequest request, HttpServletRespons
 					
 					// add encounters if this user can view them
 					if (Context.hasPrivilege(PrivilegeConstants.GET_ENCOUNTERS)) {
-						model.put("patientEncounters", Context.getEncounterService().getEncountersByPatient(p));
+						model.put(
+						    "patientEncounters",
+						    Context.getEncounterService().getEncounters(p.getPatientIdentifier().getIdentifier(), 0, 100,
```

Comment:
@ODORA0 why are you hardcoding the start(0) and length(100) ? My understanding of these two variables is that they can help paginate the results, the `start` being the page and the `length` the number of records per page.

Reply:
My bad here, I might have forgotten to make the change but changed so that we using the proper pagination parameters using the configured page size in the `maximumNumberToShow` GP or default to only 50 when empty and allow users to paginate through encounters.